### PR TITLE
Add unit test for the PolyMod method in b(l)ech32

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: src/test/bech32_tests.cpp [hunk 1]
```diff
@@ -66,4 +66,22 @@ BOOST_AUTO_TEST_CASE(bip173_testvectors_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(bech32_polymod_sanity)
```
Comment (instagibbs) [inline src/test/bech32_tests.cpp]: this a suitable upstream test?

**Comment by instagibbs** [comment]
utACK https://github.com/ElementsProject/elements/pull/595/commits/0de2b056d12f0aaa74a7cca1a74e320a9de03815

**Comment by stevenroose** [comment]
Pushed two more commits with a fix.

File: src/test/bech32_tests.cpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <bech32.h>
+#include <bech32.cpp>
```
Comment (sipa) [inline src/test/bech32_tests.cpp]: Yuck (including a .cpp file).

Comment (instagibbs) [reply src/test/bech32_tests.cpp]: *chants* export export export

Comment (instagibbs) [reply src/test/bech32_tests.cpp]: linter doesn't like it either :D

Comment (stevenroose) [reply src/test/bech32_tests.cpp]: Yeah was to test the internal methods. Changed that meanwhile.

**Comment by instagibbs** [comment]
ACK https://github.com/ElementsProject/elements/pull/595/commits/4a218cd0efdf300471341cdced18968f944578f2

**Pull request merged**